libs/packages/sam-formly/src/lib/formly-stepper/sds-stepper.ts

```diff
@@ -305,27 +312,28 @@ export class SdsStepper {
     if (this.linear) {
       this.evaluateIncompleteForms();
     }
-
-    this.changeStep(this.currentStepId).finally(() => {
-      this.activatedRoute.queryParams.subscribe(queryParam => {
-        if (queryParam[this.queryParamKey] && queryParam[this.queryParamKey] != this.currentStepId) {
-          this.changeStep(queryParam[this.queryParamKey]);
-        }
+    if (this.isRouteEnabled) {
+      this.changeStep(this.currentStepId).finally(() => {
+        this.activatedRoute.queryParams.subscribe(queryParam => {
```

Comment:
Here, move the isRouteEnabled condition check inside the `finally` block. The call to changeStep should happen regardless of `isRouteEnabled` value, but the queryParams subscription should only happen if routing is enabled